lib/src/clojure_lsp/feature/stubs.clj

```diff
@@ -48,19 +48,14 @@
   (let [result (shared/logging-time
                  "Stubs analyzed, took %s."
                  (lsp.kondo/run-kondo-on-paths! dirs true db*))
-        kondo-analysis (-> (:analysis result)
-                           (dissoc :namespace-usages :var-usages))
-        analysis (->> kondo-analysis
-                      (lsp.kondo/normalize-analysis true)
-                      (group-by :filename))]
-    (loop [state-db @db*]
-      (when-not (compare-and-set! db* state-db (update state-db :analysis merge analysis))
-        (logger/warn "Analyzis divergent from stub analysis, trying again...")
-        (recur @db*)))
+        result (-> result
+                   (update :analysis dissoc :namespace-usages :var-usages)
+                   (db/normalize-kondo {:external? true}))]
+    (swap! db* db/merge-kondo-results result)
```

Comment:
This is another place where I replaced a `loop + compare-and-set!` with `swap!`. See above for why I think this is safe.